dagmc/dagnav.py

```diff
@@ -420,53 +446,54 @@ def groups(self) -> list[Group]:
         """Get list of groups containing this volume."""
         return [group for group in self.model.groups.values() if self in group]
 
+    def _material_group(self):
+        for group in self.groups:
+            if self in group and group.name.startswith("mat:"):
+                return group
+        return None
+    
     @property
     def material(self) -> Optional[str]:
         """Name of the material assigned to this volume."""
-        for group in self.groups:
-            if self in group and group.name.startswith("mat:"):
-                return group.name[4:]
+        group = self._material_group
+        if group is not None:
+            return group.name[4:]
```

Comment:
Along the lines of the comment above, we might require some more sophisticated parsing here if the material assignment doesn't come first in the group name. This is a pre-existing issue though, so I'll note it and we can clean it up in a subsequent PR.